mingw.thread.h

```diff
@@ -20,6 +20,9 @@
 #ifndef WIN32STDTHREAD_H
 #define WIN32STDTHREAD_H
 
+//  Use the standard classes for std::, if available.

```

Comment:
I operate under the assumption that a user who includes mingw.thread.h desires that the namespace std contain the standard threading classes.
- If those are not already provided by the compiler, the mingw-std-thread library adds them.
- If those are already provided by the compiler, then we cannot both place the mingw-std-thread library's classes into std and have them yield to those of <thread>. The namespace inclusion trick that would have allowed this is foiled by the common command `using namespace std;`.
  - If we `#include <thread>`, then a user need only `#include "mingw.thread.h`, regardless of which compiler is used.
 - If we do not `#include <thread>`, then a user must independently specify the use of <thread> if their code is used with a different compiler.

In short, using `#include <thread>` in the mingw-std-thread library doesn't hurt, and simplifies the experience of the end user.

Reply:
Ok lets leave it like that, but generally I would avoid doing things behind the scenes that the user may not intend to do, for whatever reason, even if it would save a line or two of code.